test/image/mocks/bar_text_relative.json

```diff
@@ -0,0 +1,40 @@
+{
+    "data":[
+        {
+            "width":[1,0.8,0.6,0.4],
+            "text":[1,2,3,4],
+	    "textposition":"outside",
+            "y":[1,2,3,4],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "width":[0.4,0.6,0.8,1],
+            "text":["Three",2,"This bar is size one",0],
+	    "textposition":"auto",
+            "y":[3,2,1,0],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "width":1,
+            "text":[-1,-3,-2,-4],
+	    "textposition":"inside",
+            "y":[-1,-3,-2,-4],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "text":[0,"very small size",-3,-2],
```

Comment:
@alexcjohnson Without going into the details, the idea is that if a text doesn't fit inside a bar, we shrink the text by computing a scaling factor (smaller than 1).

To avoid shrinking the text unnecessarily or too much, the text is rotated 90 degrees when this results in a larger scaling factor (still smaller than 1).

---

I'm not sure if I did a good job with the above explanation. If not, I can give an example.